aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthService.kt

```diff
@@ -17,33 +17,57 @@ package com.amplifyframework.auth.cognito
 import aws.smithy.kotlin.runtime.client.endpoints.Endpoint
 import aws.smithy.kotlin.runtime.client.endpoints.EndpointProvider
+import aws.smithy.kotlin.runtime.http.interceptors.HttpInterceptor
 import com.amplifyframework.statemachine.codegen.data.AuthConfiguration
 
 interface AWSCognitoAuthService {
-    var cognitoIdentityProviderClient: CognitoIdentityProviderClient?
-    var cognitoIdentityClient: CognitoIdentityClient?
+    val cognitoIdentityProviderClient: CognitoIdentityProviderClient?
+    val cognitoIdentityClient: CognitoIdentityClient?
+    val customUserAgentPairs: MutableMap<String, String>
 
     companion object {
         internal fun fromConfiguration(configuration: AuthConfiguration): AWSCognitoAuthService {
+            val customPairs: MutableMap<String, String> = mutableMapOf()
             val cognitoIdentityProviderClient = configuration.userPool?.let { it ->
 
                 CognitoIdentityProviderClient {
                     this.region = it.region
                     this.endpointProvider = it.endpoint?.let { endpoint ->
                         EndpointProvider { Endpoint(endpoint) }
                     }
+                    this.interceptors += object : HttpInterceptor {
+                        override suspend fun modifyBeforeSerialization(context: RequestInterceptorContext<Any>): Any {
+                            customPairs.forEach { (key, value) ->
+                                context.executionContext.customUserAgentMetadata.add(key, value)
+                            }
+                            return super.modifyBeforeSerialization(context)
+                        }
+                    }
                 }
             }
 
             val cognitoIdentityClient = configuration.identityPool?.let { it ->
-                CognitoIdentityClient { this.region = it.region }
+                CognitoIdentityClient {
+                    this.region = it.region
+                    this.interceptors += object : HttpInterceptor {
+                        override suspend fun modifyBeforeSerialization(context: RequestInterceptorContext<Any>): Any {
+                            customPairs.forEach { (key, value) ->
+                                context.executionContext.customUserAgentMetadata.add(key, value)
+                            }
+                            return super.modifyBeforeSerialization(context)
+                        }
+                    }
+                }
             }
 
             return object : AWSCognitoAuthService {
-                override var cognitoIdentityProviderClient = cognitoIdentityProviderClient
-                override var cognitoIdentityClient = cognitoIdentityClient
+                override val cognitoIdentityProviderClient = cognitoIdentityProviderClient
+                override val cognitoIdentityClient = cognitoIdentityClient
+                override val customUserAgentPairs = customPairs
```

Comment:
Once the custom pairs are added, would any API call from cognito plugin also send these custom pairs?

Reply:
Yes, all Auth calls will include these pairs.  This is also in parity with JS